src/nvim/file_search.c

```diff
@@ -1396,7 +1396,12 @@ char *find_file_in_path_option(char *ptr, size_t len, int options, int first, ch
     // filename on the first call.
     if (first == true) {
       if (path_with_url(*file_to_find)) {
-        file_name = xstrdup(*file_to_find);
+        if (strncmp(*file_to_find, "file", 4) == 0) {
```

Comment:
`strncmp` avoids false positives, but it doesn't hurt to be extra clear for others reading the code.

```suggestion
        if (strncmp(*file_to_find, "file:/", 6) == 0) {
```